File: ruby/lib/onvo/questions.rb
# frozen_string_literal: true

require 'uri'
require_relative '../resource'

# Dashboard Question endpoints
class Questions < Resource
  # list questions from a dashboard
  #
  # @param dashboard_id [String] ID of the dashboard
  # @return [Array] list of questions in the given dashboard
  def list(dashboard_id)
    base_get('/questions', query: { dashboard: dashboard_id })
  end

  # create a question in a dashboard
  #
  # @param dashboard_id [String] ID of the dashoard to create the question in
  # @param query [String] the answer to the question asked
  def create(dashboard_id, query)
    base_post('/questions', body: { dashboard: dashboard_id, messages: [{ role: 'user', content: query }] })
  end

  # delete a question
  #
  # @param id [String] ID of the question to delete
  # @return [Hash] status of delete. Returns {"success": true} if successful.
  def delete(id)
    base_delete("/questions/#{id}")
  end

  # TODO : Add update method once tested on postman
end

File: python/onvo/resources/questions.py
from onvo.base_resource import BaseResource


class Questions(BaseResource):
    """Endpoints to manage Onvo questions"""

    def list(self, dashboard_id: str):
        """List all questions linked to a dashboard

        Args:
            dashboard_id (str): The dashboard id to list questions for

        Returns:
            list: A list of all the questions
        """
        return self.base_get("/questions", params={"dashboard": dashboard_id})

    def delete(self, id: str):
        """Delete an existing question

        Args:
            id (str): The question id to delete

        Returns:
            dict: A dictionary that shows status of the delete. Returns {"success": true} if successful.
        """
        return self.base_delete(f"/questions/{id}")

    def create(self, dashboard_id: str, query: str):
        """Create a new question

        Args:
            dashboard_id (str): The dashboard id to create the question for
            query (str): The query string to ask a question

        Returns:
            dict: A dictionary containing the details of the newly created question
        """
        return self.base_post(
            "/questions",
            data={
                "dashboard": dashboard_id,
                "messages": [{"role": "user", "content": query}],
            },
        )

    # TODO: create the update method once the postman shows viable example
